### PR TITLE
Fixes #565 and probably #642

### LINKED ISSUE #565: Failed transformer creates dead zone
Attempting to place a HV to MV transformer will cause the transformer to drop when inserting the 4 MV wires to step down. As a result, literally anything you place in that square will become deleted. This includes things such as flowing water, piston heads, and any block you place.

To make matters stranger, any electrical component placed on the dead block will operate for exactly 1 tick before being deleted and dropped as an item.

This effect will eventually dissipate, presumably since some error checking deletes the fake object.

### PR DESCRIPTION
**Pull request by cm0x4D**
This change is [Reviewable](https://reviewable.io/reviews/electrical-age/electricalage/644)

### CODE REVIEW

File: src/main/java/mods/eln/node/NodeBase.java [hunk 1]
```diff
@@ -153,6 +153,10 @@ public NodeBase() {
 
     boolean destructed = false;
```
Comment (Baughn) [inline src/main/java/mods/eln/node/NodeBase.java:154]: I'm a bit leery of this sort of patch, so let me think out loud.

It'll *work*, no doubt about that, but only so long as a bunch of special cases are caught. What's actually going on is... the node is getting reconnected after it's been supposedly destroyed, when the UI closes. A symptom of the simulation existing in a parallel world from Minecraft, I suppose; the block is gone, but the node is not.

Ideally we'd tie them a bit more closely together. How often is connectJob called? Often enough... checking for the block's existence every time it's run might degrade performance, or have other problems.

Maybe we could add a process that, at a relatively slow pace, checks each node to make sure it still exists in-world? That would also catch other causes of this sort of thing. It should cry havoc to the console if it finds anything, of course, but it's a fallback and debugging mechanism.

**Review by Baughn — APPROVED** [review]
This looks like it'll work, though.

File: src/main/java/mods/eln/node/NodeBase.java (hunk 1 above)
Comment (cm0x4D) [reply src/main/java/mods/eln/node/NodeBase.java:154]: > It'll work, no doubt about that, but only so long as a bunch of special cases are caught. What's actually going on is... the node is getting reconnected after it's been supposedly destroyed, when the UI closes. A symptom of the simulation existing in a parallel world from Minecraft, I suppose; the block is gone, but the node is not.

The parallel existence is not optional. It is required as nodes which's blocks are not actually loaded have to be simulated too. The *destructed* attribute was already present before, imho this is the least intrusive way to fix the issue by just using it as how it was actually intended.

> Ideally we'd tie them a bit more closely together. How often is connectJob called? Often enough... checking for the block's existence every time it's run might degrade performance, or have other problems.

This method is only called when there is an interaction of the user with the block or one of it's neighbors or a block gets destroyed. I would not call this often. Further the performance impact of reading this boolean should be not significant at all.

I can remove the check in the connectJob() method, it is just a second test that might catch (or not) other issues, the inventory issue should be already fixed in the respective Inventory classes. But for the actual costs I think it is worth to keep it.

> Maybe we could add a process that, at a relatively slow pace, checks each node to make sure it still exists in-world? That would also catch other causes of this sort of thing. It should cry havoc to the console if it finds anything, of course, but it's a fallback and debugging mechanism.

Imho maybe additionally having such a test in debug mode would make sense to find problems, but I think the impact to the performance could be significant, as this check might force Minecraft to load offline chunks just to do the check.

**Comment by Baughn** [comment]
Oh, I didn't mean that reading the boolean would be expensive. It's reading the Minecraft world to check if the block still exists that would be expensive.

As to scanning... Sure, but nothing is going to change in an unloaded chunk. It'd work well enough even if it skips all unloaded blocks.

**Comment by cm0x4D** [comment]
> Oh, I didn't mean that reading the boolean would be expensive. It's reading the Minecraft world to check if the block still exists that would be expensive.

NodeBase is not part of the Minecraft world, it is part of the EA parallel world. So there is no reading from the Minecraft world at all. 

This list of nodes (NodeManager) seem to work pretty well. I never found a node in the manager that was not destroyed when the block was. Even with the GUI bug, the block ends up destroyed. 

The problem was that by adding the IProcess objects to the simulator (in the connectJob() method), they continue to exist even after  the node instance was destroyed, as the simulator's list retains them.

> As to scanning... Sure, but nothing is going to change in an unloaded chunk. It'd work well enough even if it skips all unloaded blocks.

Ok, then it would make sense. Maybe we could add a setting to turn it on or off?

**Comment by Baughn** [comment]
I'll add the scanner as a follow-up.

**Pull request merged**